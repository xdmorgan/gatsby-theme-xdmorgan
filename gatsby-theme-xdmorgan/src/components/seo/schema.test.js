import getSchema from "./schema";

const site = {
  organization: "Dan Morgan",
  description: "Front End Engineer in Washington D.C.",
  logo: "https://danny.codes/icons/icon-256x256.png",
  title: "danny.codes",
  url: "https://danny.codes"
};

describe("JSON-LD Schemas", () => {
  test("The homepage has only website schema", () => {
    const options = {
      page: {
        description: "Front End Engineer in Washington D.C.",
        image: "https://danny.codes/icons/icon-512x512.png",
        title: "Home",
        url: "https://danny.codes"
      },
      site
    };

    const schema = getSchema(options);
    expect(schema).toHaveLength(1);
    expect(schema[0]["@type"]).toBe("WebSite");
    expect(schema).toMatchSnapshot();
  });
  test("An internal page has website and breadcrumb schemas", () => {
    const options = {
      page: {
        description: "This is an internal page description",
        image: "https://danny.codes/icons/icon-512x512.png",
        title: "About",
        url: "https://danny.codes/about"
      },
      site
    };

    const schema = getSchema(options);
    expect(schema).toHaveLength(2);
    expect(schema[0]["@type"]).toBe("WebSite");
    expect(schema[1]["@type"]).toBe("BreadcrumbList");
    expect(schema).toMatchSnapshot();
  });
  test("An editorial content page has website, breadcrumb, and blog post schemas", () => {
    const options = {
      page: {
        description: "This is a mock blog post description",
        image: "https://danny.codes/icons/icon-512x512.png",
        title: "Hello, world!",
        url: "https://danny.codes/blog/hello-world",
        blogPost: {
          author: "Dan Morgan",
          datePublished: "2019/01/21"
        }
      },
      site
    };

    const schema = getSchema(options);
    expect(schema).toHaveLength(3);
    expect(schema[0]["@type"]).toBe("WebSite");
    expect(schema[1]["@type"]).toBe("BreadcrumbList");
    expect(schema[2]["@type"]).toBe("BlogPosting");
    expect(schema).toMatchSnapshot();
  });
});
