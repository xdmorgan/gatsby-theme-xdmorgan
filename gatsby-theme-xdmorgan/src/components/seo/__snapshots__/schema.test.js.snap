// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON-LD Schemas An editorial content page has website, breadcrumb, and blog post schemas 1`] = `
Array [
  Object {
    "@context": "http://schema.org",
    "@type": "WebSite",
    "alternateName": "danny.codes",
    "image": Object {
      "@type": "ImageObject",
      "url": "https://danny.codes/icons/icon-256x256.png",
    },
    "name": "Hello, world!",
    "url": "https://danny.codes/blog/hello-world",
  },
  Object {
    "@context": "https://danny.codes",
    "@type": "BreadcrumbList",
    "itemListElement": Array [
      Object {
        "@type": "ListItem",
        "item": Object {
          "@id": "https://danny.codes/blog/hello-world",
          "image": "https://danny.codes/icons/icon-512x512.png",
          "name": "Hello, world!",
        },
        "position": 1,
      },
    ],
  },
  Object {
    "@context": "https://danny.codes",
    "@type": "BlogPosting",
    "alternateName": "danny.codes",
    "author": Object {
      "@type": "Person",
      "name": "Dan Morgan",
    },
    "datePublished": "2019/01/21",
    "description": "This is a mock blog post description",
    "headline": "Hello, world!",
    "image": Object {
      "@type": "ImageObject",
      "url": "https://danny.codes/icons/icon-512x512.png",
    },
    "mainEntityOfPage": Object {
      "@id": "https://danny.codes",
      "@type": "WebSite",
    },
    "name": "Hello, world!",
    "publisher": Object {
      "@type": "Organization",
      "logo": "https://danny.codes/icons/icon-256x256.png",
      "name": "Dan Morgan",
      "url": "https://danny.codes",
    },
    "url": "https://danny.codes/blog/hello-world",
  },
]
`;

exports[`JSON-LD Schemas An internal page has website and breadcrumb schemas 1`] = `
Array [
  Object {
    "@context": "http://schema.org",
    "@type": "WebSite",
    "alternateName": "danny.codes",
    "image": Object {
      "@type": "ImageObject",
      "url": "https://danny.codes/icons/icon-256x256.png",
    },
    "name": "About",
    "url": "https://danny.codes/about",
  },
  Object {
    "@context": "https://danny.codes",
    "@type": "BreadcrumbList",
    "itemListElement": Array [
      Object {
        "@type": "ListItem",
        "item": Object {
          "@id": "https://danny.codes/about",
          "image": "https://danny.codes/icons/icon-512x512.png",
          "name": "About",
        },
        "position": 1,
      },
    ],
  },
]
`;

exports[`JSON-LD Schemas The homepage has only website schema 1`] = `
Array [
  Object {
    "@context": "http://schema.org",
    "@type": "WebSite",
    "alternateName": "danny.codes",
    "image": Object {
      "@type": "ImageObject",
      "url": "https://danny.codes/icons/icon-256x256.png",
    },
    "name": "Home",
    "url": "https://danny.codes",
  },
]
`;
